src/producer.rs

```diff
@@ -631,6 +633,29 @@ impl<H: BuildHasher> Partitioner for DefaultPartitioner<H> {
     }
 }
 
+/// An object which represents the response sent back by the Kafka broker
+/// to confirm delivery of producer messages.
+#[derive(Debug)]
+pub struct ProduceConfirm {
+    /// The topic the messages were sent to.
+    pub topic: String,
+
+    /// The list of individual confirmations for each offset and partition.
+    pub topic_partition_confirms: Vec<ProduceTopicPartitionConfirm>,
+}
+
+/// An object which represents the broker's confirmation of received producer
+/// messages for a particular topic.
+#[derive(Debug)]
+pub struct ProduceTopicPartitionConfirm {
+    /// The offset assigned to the first message in the message set appended
+    /// to this partition, or an error if one occurred.
+    pub offset: std::result::Result<i64, error::KafkaCode>,
+
+    /// The partition to which the message(s) were appended.
+    pub partition: i32,
+}
+
```

Comment:
Yeah, I think either would work. `ProduceConfirm` and `ProducePartitionConfirm` seems fine.